arangod/Cluster/ClusterInfo.cpp

```diff
@@ -3427,15 +3437,20 @@ Result ClusterInfo::createCollectionsCoordinator(
                     << "Did not find shard in _shardServers: "
                     << p.key.copyString()
                     << ". Maybe the collection is already dropped.";
-                *errMsg =
-                    "Error in creation of collection: " + p.key.copyString() +
-                    ". Collection already dropped. " + __FILE__ + ":" +
-                    std::to_string(__LINE__);
-                dbServerResult->store(
-                    TRI_ERROR_CLUSTER_COULD_NOT_CREATE_COLLECTION,
-                    std::memory_order_release);
-                TRI_ASSERT(info.state != ClusterCollectionCreationState::DONE);
-                info.state = ClusterCollectionCreationState::FAILED;
+                report->doUnderLock([&p,
+                                     &info](CreateCollectionReport& report) {
+                  report.errMsg =
+                      "Error in creation of collection: " + p.key.copyString() +
+                      ". Collection already dropped. " + __FILE__ + ":" +
+                      std::to_string(__LINE__);
```

Comment:
You can create string before lock acquire

Reply:
It will simpler

Reply:
I want to keep impact on this fix as minimal as possible, so everything stays as it was before.